dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchServiceCollectionExtensions.cs

```diff
@@ -9,260 +9,327 @@
+    /// <param name="options">Optional options to further configure the <see cref="AzureAISearchVectorStore"/>.</param>
+    /// <param name="lifetime">The service lifetime for the store. Defaults to <see cref="ServiceLifetime.Singleton"/>.</param>
+    /// <returns>Service collection.</returns>
+    [RequiresUnreferencedCode(DynamicCodeMessage)]
+    [RequiresDynamicCode(UnreferencedCodeMessage)]
+    public static IServiceCollection AddKeyedAzureAISearchVectorStore(
+        this IServiceCollection services,
+        object? serviceKey,
+        Uri endpoint,
+        TokenCredential tokenCredential,
+        AzureAISearchVectorStoreOptions? options = default,
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
     {
         Verify.NotNull(endpoint);
         Verify.NotNull(tokenCredential);
 
-        services.AddKeyedSingleton<VectorStore>(
-            serviceId,
-            (sp, obj) =>
-            {
-                options ??= sp.GetService<AzureAISearchVectorStoreOptions>() ?? new()
-                {
-                    EmbeddingGenerator = sp.GetService<IEmbeddingGenerator>()
-                };
-                var searchClientOptions = BuildSearchClientOptions(options?.JsonSerializerOptions);
-                var searchIndexClient = new SearchIndexClient(endpoint, tokenCredential, searchClientOptions);
-
-                // Construct the vector store.
-                return new AzureAISearchVectorStore(searchIndexClient, options);
-            });
-
-        return services;
+        return AddKeyedAzureAISearchVectorStore(services, serviceKey, sp =>
+        {
+            var searchClientOptions = BuildSearchClientOptions(options?.JsonSerializerOptions);
+            return new SearchIndexClient(endpoint, tokenCredential, searchClientOptions);
+        }, _ => options!, lifetime);
     }
 
     /// <summary>
-    /// Register an Azure AI Search <see cref="VectorStore"/> with the provided <see cref="Uri"/> and <see cref="AzureKeyCredential"/> and the specified service ID.
+    /// Registers a <see cref="AzureAISearchVectorStore"/> as <see cref="VectorStore"/>
+    /// using the provided <paramref name="endpoint"/> and <paramref name="keyCredential"/>.
     /// </summary>
-    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="VectorStore"/> on.</param>
+    /// <inheritdoc cref="AddKeyedAzureAISearchVectorStore(IServiceCollection, object?, Uri, AzureKeyCredential, AzureAISearchVectorStoreOptions?, ServiceLifetime)"/>
+    [RequiresUnreferencedCode(DynamicCodeMessage)]
+    [RequiresDynamicCode(UnreferencedCodeMessage)]
+    public static IServiceCollection AddAzureAISearchVectorStore(
+        this IServiceCollection services,
+        Uri endpoint,
+        AzureKeyCredential keyCredential,
```

Comment:
Same as above, but using `AzureKeyCredential` rather than `TokenCredential` (I kept it for backward copat and ease of use)